lms/views/lti/basic_launch.py

```diff
@@ -82,33 +82,37 @@ def unconfigured_launch(self):
 
         This happens if we cannot resolve a document URL for any reason.
         """
+        return self._configure_js_for_file_picker()
 
-        if not self.request.has_permission(Permissions.LTI_CONFIGURE_ASSIGNMENT):
-            # Looks like the user is not an instructor, so show an error page
-
-            # https://docs.pylonsproject.org/projects/pyramid/en/latest/narr
-            # /renderers.html?highlight=override_renderer#overriding-a-renderer-at-runtime
-            self.request.override_renderer = "lms:templates/lti/basic_launch/unconfigured_launch_not_authorized.html.jinja2"
-            return {}
-
-        form_fields = {
-            param: value
-            for param, value in self.request.lti_params.items()
-            # Don't send over auth related params. We'll use our own
-            # authorization header
-            if param
-            not in ["oauth_nonce", "oauth_timestamp", "oauth_signature", "id_token"]
-        }
-
-        form_fields["authorization"] = BearerTokenSchema(
-            self.request
-        ).authorization_param(self.request.lti_user)
+    @view_config(
+        route_name="lti.reconfigure",
+        renderer="json",
+    )
+    def reconfigure_assignment(self):
+        """
+        Show the file-picker for the user to choose a document.
 
-        self.context.js_config.enable_file_picker_mode(
-            form_action=self.request.route_url("configure_assignment"),
-            form_fields=form_fields,
+        This happens if we cannot resolve a document URL for any reason.
+        """
+        assignment = self.assignment_service.get_assignment(
+            tool_consumer_instance_guid=self.request.lti_params[
+                "tool_consumer_instance_guid"
+            ],
+            resource_link_id=self.request.lti_params.get("resource_link_id"),
         )
-        return {}
+        self._configure_js_for_file_picker()
+        return {
+            # Info about the assignments current configuration
+            "assignment": {
+                "id": assignment.id,
+                "group_set_id": assignment.extra.get("group_set_id"),
+                "document": {
+                    "url": assignment.document_url,
+                },
+            },
+            # Data needed to re-configure it
+            "filePicker": self.context.js_config._config["filePicker"],
```

Comment:
Getting the config that we set as a side effect with `self._configure_js_for_file_picker()`